http-clients/src/main/java/org/triplea/http/client/SystemIdHeader.java

```diff
@@ -0,0 +1,66 @@
+package org.triplea.http.client;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.prefs.BackingStoreException;
+import java.util.prefs.Preferences;
+import javax.annotation.Nullable;
+import lombok.AccessLevel;
+import lombok.Setter;
+import lombok.experimental.UtilityClass;
+import lombok.extern.java.Log;
+
+/**
+ * Creates headers with a user's 'system-id'. If a system-id has not been generated, one will be
+ * generated and stored in preferences.
+ */
+@Log
+@UtilityClass
+public class SystemIdHeader {
+  public static final String SYSTEM_ID_HEADER = "system-id-header";
```

Comment:
I'm not sure about the `TripoleA` prefix:

> In short, although in theory the "X-" convention was a good way to
   avoid collisions (and attendant interoperability problems) between
   standardized parameters and unstandardized parameters, in practice
   the benefits have been outweighed by the costs associated with the
   leakage of unstandardized parameters into the standards space.


>   Makes specific recommendations about how to proceed in a world
       without the distinction between standardized and unstandardized
       parameters (although only for parameters with textual names, not
       parameters that are expressed as numbers, which are out of the
       scope of this document).

For better or worse, we're doing security through obfuscation with this parameter, it might be better to not call it out as TripleA specific.

Reply:
If you're going for the obfuscation route it might acutally make sense to embed this information in the auth token somehow

Reply:
Well, system-id is the new hashed-mac. We're a bit more secure at least as it is not deterministic to someone else that can determine your mac and secondly we would no longer be leaking hashed-mac + ip to chatters.

How would it look to combine system-id with login token? Also to consider is that system-id is passed as parameter (header) to login to later determine if login is even allowed. Are you thinking an HMAC like situation? Would that help given if someone has altered their system-id at login, does it matter if they alter it later (considering it's already been altered)

Reply:
That's thr question, I guess it depends if you want to makr faking it a little bit more annoying